xfrd.c

```diff
@@ -1965,16 +1986,15 @@ xfrd_parse_received_xfr_packet(xfrd_zone_type* zone, buffer_type* packet,
 				zone->apex_str, zone->master->ip_address_spec));
 			region_destroy(tempregion);
 			return xfrd_packet_bad;
-		}
-		if(zone->soa_disk_acquired != 0 && zone->soa_disk.serial == soa->serial) {
-			DEBUG(DEBUG_XFRD,1, (LOG_INFO, "xfrd: zone %s got "
-						       "update indicating "
-						       "current serial",
+		} else if(compare_serial(ntohl(soa->serial),
+		                         ntohl(last_soa->serial)) == 0)
```

Comment:
Also this call to compare_serial() is not needed.  It could be: ntohl(soa->serial) == ntohl(..)